user_tools/src/spark_rapids_pytools/cloud_api/sp_types.py

```diff
@@ -727,10 +729,15 @@ def _set_remaining_configuration_list(self) -> None:
                                                         sectionKey=self.ctxt.get(config_file_section[config_file]))
                 if loaded_conf_dict:
                     self.ctxt.update(loaded_conf_dict)
-            for prop_elem in properties_map_arr:
-                if loaded_conf_dict and prop_elem.get('propKey') not in loaded_conf_dict:
+            # If the property key is not already set, below code attempts to set the property
+            # using an environment variable. This is a fallback mechanism to populate configuration
+            # properties from the environment if they are not already set in the
+            # loaded configuration.
+            for prop_entry in properties_map_arr:
+                prop_entry_key = prop_entry.get('propKey')
+                if self.ctxt.get(prop_entry_key) is None:
```

Comment:
This check `if self.ctxt.get(prop_entry_key) is None:` is already in function `_set_env_prop_from_env_var` definition. We can probably remove one of them.

Reply:
Thanks @cindyyuanjiang! Removed the check here.